modules/client-handler/src/test/java/org/apache/ignite/client/handler/JdbcQueryCursorHandlerImplTest.java

```diff
@@ -0,0 +1,137 @@
+        ClientResource rsrc = mock(ClientResource.class);
+
+        JdbcQueryCursorHandler cursorHandler = new JdbcQueryCursorHandlerImpl(resourceRegistryMocked);
+
+        when(resourceRegistryMocked.get(anyLong())).thenAnswer(new Answer<ClientResource>() {
+            @Override
+            public ClientResource answer(InvocationOnMock invocation) {
+                return rsrc;
+            }
+        });
+
+        when(rsrc.get(AsyncSqlCursor.class)).thenAnswer(new Answer<AsyncSqlCursor<InternalSqlRow>>() {
+            @Override
+            public AsyncSqlCursor<InternalSqlRow> answer(InvocationOnMock invocation) {
+                return new AsyncSqlCursor<>() {
+                    @Override
+                    public SqlQueryType queryType() {
+                        throw new UnsupportedOperationException("queryType");
+                    }
+
+                    @Override
+                    public ResultSetMetadata metadata() {
+                        throw new UnsupportedOperationException("metadata");
+                    }
+
+                    @Override
+                    public boolean hasNextResult() {
+                        return true;
+                    }
+
+                    @Override
+                    public CompletableFuture<Void> onClose() {
+                        return null;
+                    }
+
+                    @Override
+                    public CompletableFuture<Void> onFirstPageReady() {
+                        return null;
+                    }
+
+                    @Override
+                    public CompletableFuture<AsyncSqlCursor<InternalSqlRow>> nextResult() {
+                        if (nextResultThrow) {
+                            throw new SqlException(STMT_PARSE_ERR, new Exception("nextResult exception"));
+                        } else {
+                            AsyncSqlCursorImpl<InternalSqlRow> sqlCursor = mock(AsyncSqlCursorImpl.class);
+
+                            lenient().when(sqlCursor.requestNextAsync(anyInt()))
+                                    .thenAnswer((Answer<BatchedResult<InternalSqlRow>>) invocation -> {
+                                        throw new IgniteInternalException(VALIDATION_ERR, "requestNextAsync error");
+                                    }
+                            );
+
+                            return CompletableFuture.completedFuture(sqlCursor);
+                        }
+                    }
+
+                    @Override
+                    public CompletableFuture<BatchedResult<InternalSqlRow>> requestNextAsync(int rows) {
+                        return nullCompletedFuture();
+                    }
+
+                    @Override
+                    public CompletableFuture<Void> closeAsync() {
+                        return nullCompletedFuture();
+                    }
+                };
+            }
+        });
+
+        try {
```

Comment:
what reason to have the try/catch block?

Reply:
look, this test check that no exception is raised from JdbcQueryCursorHandlerImpl#getMoreResultsAsync
comment exception handling in mention method and try to run test.